### PR TITLE
lib/syncthing: Add wrapper for access to model

### PR DESCRIPTION
**Pull request by pixelspark**
### Purpose

Wrap access to Model for users that use the syncthing Go package. See discussion: https://github.com/syncthing/syncthing/pull/9619#pullrequestreview-2212484910

### Testing

It works with the iOS app. Other than that, there are no current users of this API (to my knowledge) as Model was only exposed recently form the iOS app.

### Screenshots

n/a

### Documentation

n/a (not user visible)

## Authorship

Already in AUTHORS

### CODE REVIEW

**Comment by pixelspark** [comment]
@imsodin I have moved the wrapper over to lib/syncthing as you suggested. 

I have yet to look into narrowing the scope of this wrapper (i.e. as discussed use FolderSummaryService from the iOS app would remove the need to expose DBSnapshot). 

My suggestion would be to iterate, i.e. review and merge this now, and I'll have a go at narrowing later.

**Comment by imsodin** [comment]
Iteration seems fine. Which also means whatever we merge might get released, thus could we please already use the "internal" naming that was brought up before? Basically what you call model or modelwrapper now, lets call that internal. And add a comment on it in `App`, something like this (very welcome to (heavily) change this, just what came to mind quickly): `Internal provides access to internal workings of syncthing. Unlike other exported methods on `App`, this is not intended to be generally safe to use and be stable (behaviour and API wise).`

File: lib/syncthing/model.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	model model.Model
+}
+
+func newModelWrapper(model model.Model) *ModelWrapper {
+	return &ModelWrapper{
+		model: model,
+	}
+}
+
+func (m *ModelWrapper) State(folderID string) (string, time.Time, error) {
```
Comment (imsodin) [inline lib/syncthing/model.go:32]:
```suggestion
func (m *ModelWrapper) FolderState(folderID string) (string, time.Time, error) {
```

File: lib/syncthing/model.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+func (m *ModelWrapper) BlockAvailability(folderID string, file protocol.FileInfo, block protocol.BlockInfo) ([]model.Availability, error) {
+	return m.model.Availability(folderID, file, block)
+}
+
+func (m *ModelWrapper) GlobalFileInfo(folderID, path string) (protocol.FileInfo, bool, error) {
+	return m.model.CurrentGlobalFile(folderID, path)
+}
+
+func (m *ModelWrapper) GlobalTreeEntries(folderID string, prefix string, levels int, returnOnlyDirectories bool) ([]*model.TreeEntry, error) {
```
Comment (imsodin) [inline lib/syncthing/model.go:56]: optional: The entries suffix is just filler imo.

```suggestion
func (m *ModelWrapper) GlobalTree(folderID string, prefix string, levels int, returnOnlyDirectories bool) ([]*model.TreeEntry, error) {
```

**Review by imsodin — COMMENTED** [review]
Some naming comments in the review. Few of them I feel a bit stronger about, few of them are just weekly held opinions - the latter begin with "optional:", if you disagree feel free to just e.g. downvote (to ack you saw it) and otherwise ignore, no need to respond why or anything.

Well turned out to be just two comments :)  
Otherwise looks good. Would very much like to see moving some of your utility functions into lib/syncthing and reducing the internals surface. Though also the more I look at it, the less bad it looks as is already :)

**Comment by pixelspark** [comment]
@imsodin I made some changes, how do you like it now?

**Pull request merged**